### PR TITLE
feat: allow mana as a transaction

### LINKED ISSUE #1198: Add a loading indicator and disable the authorization toggle on load
The video shows how the indicator turns to `enabled` after a few moments of being disabled

Probably more noticiable on first load.

Also, when the user wants to approve/disapprove MANA, but reject the tx, the `claim` button keeps loading forever

[toggle.zip](https://github.com/decentraland/builder/files/5675855/toggle.zip)

### LINKED ISSUE #1223: Prevent users to claim a name if he doesn't have 100 MANA

### PR DESCRIPTION
**Pull request by nicosantangelo**
Closes #1198 
Closes #1223

### CODE REVIEW

File: src/modules/ens/utils.ts [hunk 1]
```diff
@@ -63,3 +63,7 @@ export function getDomainFromName(name: string): string {
 export function getNameFromDomain(domain: string): string {
   return domain.split('.')[0]
 }
+
+export function isAllowed(allowance: string) {
+  return Number(allowance) > 100
```
Comment (nachomazzara) [inline src/modules/ens/utils.ts]: I know that this was this way before, but should we use `100 * 1e18` instead?

**Review by nachomazzara — COMMENTED** [review]
Thanks @nicosantangelo 💯

File: src/modules/ens/utils.ts [hunk 2]
```diff
@@ -64,6 +64,6 @@ export function getNameFromDomain(domain: string): string {
   return domain.split('.')[0]
 }
 
-export function isAllowed(allowance: string) {
-  return Number(allowance) > 100
+export function isEnoughClaimMana(mana: string) {
+  return Number(mana) > 100
```
Comment (nachomazzara) [inline src/modules/ens/utils.ts:68]: If the authorization and the balance of MANA is `>=`, is enough to claim a name.

Comment (nachomazzara) [reply src/modules/ens/utils.ts:68]: Also, just to not lose it (it is in an old comment). Should we need to change `100` to `100 * `1e18`?

File: src/components/ClaimENSPage/ClaimENSPage.tsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ export default class ClaimENSPage extends React.PureComponent<Props, State> {
 
   handleManaApprove = async () => {
     const { allowance, onAllowMana } = this.props
-    const manaToAllow = isAllowed(allowance) ? 0 : getMaximumValue()
+    const manaToAllow = isEnoughClaimMana(allowance) ? 0 : getMaximumValue()
```
Comment (nachomazzara) [inline src/components/ClaimENSPage/ClaimENSPage.tsx:25]: If the allowance is `> 0`, the action should be: 
1) Clean the allowance
2) Set a new allowance

I think that we should change this to:

```typscript
    const manaToAllow = allowance > 0 ? 0 : getMaximumValue()
```

File: src/components/ActivityPage/Transaction/Transaction.tsx [hunk 4]
```diff
@@ -280,7 +280,7 @@ const Transaction = (props: Props) => {
         <TransactionDetail
           address={address}
           text={
-            isAllowed(allowance) ? (
+            isEnoughClaimMana(allowance) ? (
```
Comment (nachomazzara) [inline src/components/ActivityPage/Transaction/Transaction.tsx:283]: Based on the [comment below](https://github.com/decentraland/builder/pull/1224/commits/44fddeb119dfc359c9026a016a1f3110785a558b#r547029705), we should change this check against `0` too.

**Comment by nachomazzara** [comment]
This is great @nicosantangelo!!

I think that we need to rebase it. (I'd mistakenly changed the base to master and then rollbacked it. I thought that we had merged `feat/claim-name` to master)

File: src/modules/ens/sagas.ts [hunk 5]
```diff
@@ -301,11 +314,31 @@ function* handleClaimNameRequest(action: ClaimNameRequestAction) {
...
+    const txHash: string = yield call(() =>
+      manaContract.methods
+        .approve(Address.fromString(CONTROLLER_ADDRESS), allowance)
+        .send({ from })
+        .getReceipt()
+    )
+
+    yield put(allowClaimManaSuccess(allowance, from.toString(), txHash))
+  } catch (error) {
+    const ensError: ENSError = { message: error.message }
```
[the remark this answers is not in the document]
Comment (nicosantangelo) [reply src/modules/ens/sagas.ts:357]: True!

File: src/modules/ens/utils.ts (hunk 2 above)
Comment (nicosantangelo) [reply src/modules/ens/utils.ts]: I checked and both the allowance and `wallet.mana` already had the number divided by `1e18`

**Review by nachomazzara — APPROVED** [review]
@nicosantangelo works perfect!!! 

Can you add a comment explaining why we are checking against `100` instead of `0` for cleaning authorizations?

Once we have that we can merge it 🚀

**Pull request merged**